### PR TITLE
Rescale image back if it was scaled during PIL conversion

### LINKED ISSUE #22392: Inconsistent Normalization for ViTImageProcessor when `do_resize` is False
### System Info

- `transformers` version: 4.26.1
- Platform: Linux-5.4.0-121-generic-x86_64-with-glibc2.31
- Python version: 3.10.9
- Huggingface_hub version: 0.13.2
- PyTorch version (GPU?): 2.0.0+cu117 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: no
- Using distributed or parallel set-up in script?: no

### Who can help?

@amyeroberts

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```py
from transformers import AutoImageProcessor
from PIL import Image
import torchvision.transforms as T

im = Image.open("t.png").convert("RGB")
to_tens = T.ToTensor()
extractor = AutoImageProcessor.from_pretrained("./pretrained/facebook/vit-msn-small")
print(extractor)  # Instance of ViTImageProcessor.

# When `do_resize` is True:
x1 = extractor(im, return_tensors="pt").pixel_values
x2 = extractor(to_tens(im), return_tensors="pt").pixel_values
print(abs(x2 - x1).mean()) # Close to 0; Correct.

# When `do_resize` is False:
x1 = extractor(im, return_tensors="pt", do_resize=False).pixel_values
x2 = extractor(to_tens(im), return_tensors="pt", do_resize=False).pixel_values
print(abs(x2 - x1).mean()) # Not close to 0; Differing behaviour.

# Additional multiplication of 255 to torch.Tensor input:
x1 = extractor(im, return_tensors="pt", do_resize=False).pixel_values
x2 = extractor(to_tens(im) * 255, return_tensors="pt", do_resize=False).pixel_values
print(abs(x2 - x1).mean()) # Close to 0; Correct again.
```

### Expected behavior

Currently, when `do_resize` is False, the tensor has to be multiplied by 255 first, while when `do_resize` is True, it is not needed. The behaviour should be consistent.

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

Resolves an issue that occurs, when a float image, with values between `[0-1]` is passed into an image processor, its inputs are rescaled back between `[0, 255]` to convert to a `PIL.Image.Image` for resizing and isn't rescaled again. This results in inconsistent outputs depending on whether the `do_resize` flag is `True` or `False`. 

Images of this type are typically fed in when pipelines that convert images to torch tensors using `ToTensor`. 

Fixes #22392

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
LGTM, thanks!

**Review by ydshieh — APPROVED** [review]
Thank you! 

(and thank you for the detailed comment in the original issue page. I am somehow confused until I read that comment!)
